apps/flame-defi/app/features/evm-wallet/services/non-fungible-position-manager-service/index.ts

```diff
@@ -350,6 +415,343 @@ export class NonfungiblePositionManagerService extends GenericContractService {
+      throw e;
+    }
+  }
+
+  /**
+   * Encodes a decreaseLiquidity function call to be used in a multicall transaction.
+   * @private
+   */
+  private encodeDecreaseLiquidityCall(
+    tokenId: string,
+    liquidity: bigint,
+    amount0Min: bigint,
+    amount1Min: bigint,
+    deadline: number,
+  ): HexString {
+    return encodeFunctionData({
+      abi: this.abi,
+      functionName: "decreaseLiquidity",
+      args: [
+        {
+          tokenId: BigInt(tokenId),
+          liquidity,
+          amount0Min,
+          amount1Min,
+          deadline,
+        },
+      ],
+    });
+  }
+
+  /**
+   * Encodes a collect function call to be used in a multicall transaction.
+   * @private
+   */
+  private encodeCollectCall(
+    tokenId: string,
+    recipient: Address,
+    amount0Max: bigint,
+    amount1Max: bigint,
+  ): HexString {
+    return encodeFunctionData({
+      abi: this.abi,
+      functionName: "collect",
+      args: [
+        {
+          tokenId: BigInt(tokenId),
+          recipient,
+          amount0Max,
+          amount1Max,
+        },
+      ],
+    });
+  }
+
+  /**
+   * Encodes an unwrapWETH9 function call to be used in a multicall transaction.
+   * This method is used to unwrap WETH (wrapped native token) back to the native token (e.g., TIA)
+   * @private
+   */
+  private encodeUnwrapWETHCall(
+    amountMinimum: bigint,
+    recipient: Address,
+  ): HexString {
+    return encodeFunctionData({
+      abi: this.abi,
+      functionName: "unwrapWETH9",
+      args: [amountMinimum, recipient],
+    });
+  }
+
+  /**
+   * Encodes a sweepToken function call to be used in a multicall transaction.
+   * This method sweeps all of the specified token from the contract to the recipient.
+   * @private
+   */
+  private encodeSweepTokenCall(
+    token: Address,
+    amountMinimum: bigint,
+    recipient: Address,
+  ): HexString {
+    return encodeFunctionData({
+      abi: this.abi,
+      functionName: "sweepToken",
+      args: [token, amountMinimum, recipient],
+    });
+  }
+
+  /**
+   * Extended write contract method that allows specifying gas limit
+   * @private
+   */
+  private async writeContractMethodWithGas(
+    chainId: number,
+    methodName: string,
+    args: unknown[] = [],
+    gas?: bigint,
+  ): Promise<HexString> {
+    const walletClient = await getWalletClient(this.wagmiConfig, { chainId });
+    if (!walletClient) {
+      throw new Error("No wallet client available");
+    }
+
+    try {
+      return await walletClient.writeContract({
+        address: this.contractAddress,
+        abi: this.abi,
+        functionName: methodName,
+        args,
+        gas,
+        account: walletClient.account?.address ?? null,
+      });
+    } catch (e) {
+      console.error(`Error in ${methodName}:`, e);
+      throw e;
+    }
+  }
+
+  /**
+   * Helper to get the wallet client
+   */
+  private async getWalletClient(chainId: number): Promise<WalletClient> {
```

Comment:
Don't just copy/paste these functions! Move them out of here and out of the swap router service and define them as methods on `GenericContractService` and update the services accordingly.

You can take this refactor a step further and update GenericContractService's `readContractMethod` and `writeContractMethod` to use these methods also.

Reply:
I was just focusing on getting everything working. I know there is duplication here from swap. I figured we would come back and move these out at a later time,